### PR TITLE
Feat: cli enhancement and improved deal validation

### PR DESCRIPTION
**Pull request by h4sh3d**
Allows to express lower and upper bounds for each assets on each network, used when validating maker/taker deals.

### CODE REVIEW

**Comment by h4sh3d** [comment]
Local tests shows for now some weird behavior, do not merge yet.

**Comment by h4sh3d** [comment]
This is now good for a review.  cc @TheCharlatan

File: src/config.rs [hunk 1]
```diff
@@ -260,6 +398,72 @@ pub struct ArbConfig {
...
+    pub max_amount: Option<T>,
+}
+
+impl<T> TradeableAmounts<T>
+where
+    T: FromStr + Display,
+    T::Err: Display,
+{
+    /// Map tradeable amounts into new type
+    pub fn map<U>(self) -> TradeableAmounts<U>
```
Comment (TheCharlatan) [inline src/config.rs:427]: What is this function for? I removed it and nothing broke.

File: src/config.rs [hunk 2]
```diff
@@ -260,6 +398,72 @@ pub struct ArbConfig {
...
+            max_amount: self.max_amount.map(|max| max.into()),
+        }
+    }
+
+    /// Validate a given amount based on potential rules (min, max)
+    pub fn validate_amount(&self, amount: T) -> Result<(), Error>
+    where
+        T: PartialOrd<T> + Copy,
+    {
+        use config::ConfigError::Message;
```
Comment (TheCharlatan) [inline src/config.rs:444]: Why is this imported here and nowhere else?

**Review by TheCharlatan — CHANGES_REQUESTED** [review]

File: src/config.rs (hunk 1 above)
Comment (h4sh3d) [reply src/config.rs:427]: This function was needed before to convert internal types `btc::Amount` and `xmr::Amount` into real types from `bitcoin` and `monero` crates, after 16200a24e91b4859ee1b98435d0f75b021e0abfa this function was not needed anymore but I kept it as it is a generic implementation of `map` for the tradeable amounts type, so maybe someday we'll need it again.

File: src/config.rs (hunk 2 above)
Comment (h4sh3d) [reply src/config.rs]: It was imported at two places and not used everywhere. I import them at function scope level to shorten the error return lines, as this is used elsewhere it's now imported everywhere with c003b47c433a812edb3b23359e843a9555e39bb3

File: src/config.rs [hunk 3]
```diff
@@ -189,6 +199,118 @@ impl Config {
...
+            Network::Mainnet => Some(Self::mainnet_btc_default_tradeable()),
+            _ => None,
+        }
+    }
+
+    // Helper function to return default mainnet btc tradeable amounts
+    fn mainnet_btc_default_tradeable() -> TradeableAmounts<bitcoin::Amount> {
+        use bitcoin::Amount;
+        TradeableAmounts {
+            min_amount: None,
```
[the remark this answers is not in the document]
Comment (h4sh3d) [reply src/config.rs]: I think it is a good idea to add one now. I chosen something in the same range as Monero, `0.00001 BTC` which is currently `0.17 USD`.

Done in b1eafd4845858b39091658de95bdd21538d42d50

**Comment by h4sh3d** [comment]
Thanks for the review @TheCharlatan, the result is now better (I took the opportunity to add more doc and fix my mistake in 19108957e3a642767237f7b8552faccac950a70a)

**Review by TheCharlatan — APPROVED** [review]

**Pull request merged**